WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -70,7 +70,12 @@ struct InsightStoreState {
     var topCommentsInsight: StatsCommentsInsight?
     var commentsInsightStatus: StoreFetchingStatus = .idle
 
-    var todaysStats: StatsTodayInsight?
+    var todaysStats: StatsTodayInsight? {
+        didSet {
+            TodayWidgetStats.saveData(views: todaysStats?.viewsCount ?? 0,
```

Comment:
I think we only want to save the data here if we're loading the today data for the site that's selected for the widget. I tested this and confirms it doesn't seem right:

* Choose a site (Site A), load the today widget, and check the stats
* In the app, browse to another site (Site B) and view its insights screen (loading the insights for that site, which will trigger this code)
* Switch back to the widget and you'll see Site B's stats briefly before the widget refreshes the data and fetches Site A's stats.